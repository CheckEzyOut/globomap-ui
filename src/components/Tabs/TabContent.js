import React from 'react';
import { connect } from 'react-redux';
import { registerTab, setTab } from '../../redux/modules/tabs';

class TabContent extends React.Component {

  componentDidMount() {
    this.props.registerTab(this.props.tabKey);
  }

  render() {
    return this.props.currentTab === this.props.tabKey
      ? this.props.children
      : null;
  }

}

function mapStateToProps(state) {
  return {
    currentTab: state.tabs.currentTab
  };
}

export default connect(
  mapStateToProps,
  { registerTab, setTab }
)(TabContent);
