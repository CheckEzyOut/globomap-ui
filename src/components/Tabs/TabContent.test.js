import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import { TabContent } from './TabContent';

describe("TabContent", function() {

  let componentDidMountStub;

  beforeEach(function() {
    componentDidMountStub = sinon.stub(TabContent.prototype, 'componentDidMount').value('newValue');
  });

  afterEach(function () {
    componentDidMountStub.restore();
  });

  it('should render TabContent component', () => {
    const component = shallow(<TabContent><div></div></TabContent>);
    expect(component).toMatchSnapshot();
  });

});
