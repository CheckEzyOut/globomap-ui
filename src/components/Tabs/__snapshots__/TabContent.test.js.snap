// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabContent should render TabContent component 1`] = `<div />`;
